### PR TITLE
fixed restart issue related to overwriting pk initial conditions

### PR DESCRIPTION
**Pull request by dasvyat**
Fields read from checkpoint are marked as changed to initiate a potential update of secondary fields.

### CODE REVIEW

File: src/state/Checkpoint.cc [hunk 1]
```diff
@@ -216,10 +216,21 @@ double Checkpoint::Read(State& S, const std::string& file_or_dirname)
...
-      if (domain.empty()) domain = "domain";
-
+      if (domain.empty()) domain = "domain";      
+      
       bool read_complete = field->second->ReadCheckpoint(*output_[domain]);
-      if (read_complete) field->second->set_initialized();
+      if (read_complete) {
+        if (S.HasFieldEvaluator(field->first)) {
+          Teuchos::RCP<FieldEvaluator> fm = S.GetFieldEvaluator(field->first);
+          Teuchos::RCP<PrimaryVariableFieldEvaluator> solution_evaluator =
```
Comment (lipnikov) [inline src/state/Checkpoint.cc:225]: This type of checks is not very nice. Should we add more functions to class FieldEvaluator to keep track of the evaluator type?

Comment (dasvyat) [reply src/state/Checkpoint.cc:225]: I'm not a big fan of these checks either, but let fix this problem first and then update FieldEvaluator interfaces.

Comment (lipnikov) [reply src/state/Checkpoint.cc:225]: Agree, but create a ticket before we forget.

**Comment by jd-moulton** [comment]
We just updated Alquimia and MSTK on the master branch (minor patch updates).  I merged just to force a test that builds against these new versions.

**Pull request merged**